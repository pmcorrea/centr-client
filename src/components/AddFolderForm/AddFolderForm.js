import React, { Component } from 'react'
import './AddFolderForm.css'
import MainContext from '../../contexts/MainContext';
import config from '../../config'
import TokenHelpers from '../../services/token-helpers'

export default class AddFolderForm extends Component {
  static contextType = MainContext;

	constructor(props) {
    super(props);
	this.state = {
		folder_name: '',
    }
  }

	updateFolderName(input) {
		this.setState({
			folder_name: input,
		});
	}

	handleSubmit(event) {
    	event.preventDefault();
		const folder  = this.state;

		fetch(`${config.API_ENDPOINT}/folders`, {
			method: 'POST',
			headers: {
					'content-type': 'application/json',
					'authorization': `bearer ${TokenHelpers.getAuthToken()}`,
			},
			body: JSON.stringify({
				folder
			})
		})
		.then(response => {
			if (!response.ok) {
				throw new Error(response.status)
			}
			return response.json()
		})
		.then(result => {
			// return full folder details and insert into context
			this.context.addFolder(result[0]);
		})
		.then(this.props.history.push('/'))
		.catch(error => this.setState({error}))
	}

	validateLength() {
		const folderName = this.state.name

		if (this.state.name.trim().length === 0) {
			return 'Name can not be blank'
		}

		if (folderName.length === 0) {
			return 'Name can not be blank'
		}

		if ((folderName.length < 3)&&(folderName.length > 0)) {
			return 'Name must be longer than 3 characters'
		}
	}

	validateExistingName() {
		const folderName = this.state.name
		const folders = this.context.folders.map(folder => folder.name)

		if (folders.find(folder => folder === folderName)) {
			return 'Folder name already exists'
		}
	}
  
  render(){  
    return (
    <>
    
      <form className={['Noteful-form'].join(' ')} onSubmit={e => this.handleSubmit(e)}>
        <label htmlFor='note_input_field'>Folder Name</label>
        <input id='note_input_field' type='text' onChange={e => this.updateFolderName(e.target.value)}></input>

        <button type='submit'>Submit</button>
      </form>
    </>
  )
  }
  
}
