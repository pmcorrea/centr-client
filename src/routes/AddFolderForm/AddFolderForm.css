.AddFolder__form_container {
  display: flex;
	flex-direction: column;
	width: 100vw;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  height: 100vh;
  background-color: white;
}

.AddFolder__form{
  width: 80vw;
  height: 50vh;
  margin: 0 auto;
  background-color: white;
  color: black
}

.AddFolder__input {
  border-top: 0px;
  border-right: 0px;
  border-left: 0px;
  border-bottom: 2px rgba(214, 214, 214, 0.63) solid;
  width: 100%; 
  margin-top: 10px;
  font-size: 15px;
  color: black;
  background-color: rgb(255, 255, 255);
  font-size: 15px;
	padding: 8px;
	width: 100%;
  border-radius: 0px;
  margin-top: 10px;
  margin-bottom: 30px;
}

.AddFolder__input:hover{
  border-color: orange;
}


.AddFolder__input::placeholder {
  color: gray;
  font-size: 15px;
}

.AddFolder__button {
  font-size: 15px;
	padding: 15px;
	width: 100%;
	border: 0px solid white;
  border-radius: 5px;
  margin-top: 10px;
  font-weight: bold;
  background-color: rgb(112, 39, 161);  
  color: white;
  box-shadow: 0px 5px 20px rgba(128, 128, 128, 0.733);
}

.AddFolder__button:hover{
  color: orange;
}