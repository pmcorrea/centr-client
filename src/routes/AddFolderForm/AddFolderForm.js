import React, { Component } from 'react'
import './AddFolderForm.css'
import MainContext from '../../contexts/MainContext';
import AuthApiService from "../../services/auth-api-service.js"


export default class AddFolderForm extends Component {
  static contextType = MainContext;

	constructor(props) {
    super(props);
	this.state = {
		folder_name: '',
    }
  }

	updateFolderName(input) {
		this.setState({
			folder_name: input,
		});
	}

	handleSubmit(event) {
    	event.preventDefault();
		const folder  = this.state;

		AuthApiService.postFolder(folder)
			.then(result => {
			this.context.getDataWithToken()
			this.props.history.goBack()
		})
		.catch(error => 
			error => this.setState({error})
		) 
	}

	validateLength() {
		const folderName = this.state.name

		if (this.state.name.trim().length === 0) {
			return 'Name can not be blank'
		}

		if (folderName.length === 0) {
			return 'Name can not be blank'
		}

		if ((folderName.length < 3)&&(folderName.length > 0)) {
			return 'Name must be longer than 3 characters'
		}
	}

	validateExistingName() {
		const folderName = this.state.name
		const folders = this.context.folders.map(folder => folder.name)

		if (folders.find(folder => folder === folderName)) {
			return 'Folder name already exists'
		}
	}
  
	render() {  
		return (
			<>
				<div className="AddFolder__form_container">
					<form className="AddFolder__form" onSubmit={e => this.handleSubmit(e)}>
					<input className="AddFolder__input" 
						id='post_input_field' 
						type='text' 
						placeholder="folder title" 
						onChange={e => this.updateFolderName(e.target.value)}></input>
						
					<button className="AddFolder__button" type='submit'>submit</button>
					</form>
				</div>
			</>
		)
	}
}
