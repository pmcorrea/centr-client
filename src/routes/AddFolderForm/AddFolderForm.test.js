import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddFolderForm from './AddFolderForm'

describe(`AddFolderForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.AddFolderForm by default', () => {
    const wrapper = shallow(<AddFolderForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the AddFolderForm given props', () => {
    const wrapper = shallow(<AddFolderForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
