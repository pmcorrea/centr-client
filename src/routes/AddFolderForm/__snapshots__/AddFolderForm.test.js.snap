// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolderForm component renders a form.AddFolderForm by default 1`] = `
<Fragment>
  <button
    className="go_back_button"
    onClick={[Function]}
  />
  <div
    className="AddFolder__form_container"
  >
    <form
      className="AddFolder__form"
      onSubmit={[Function]}
    >
      <button
        className="go_back_button"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className="arrow-alt-circle-left"
          fixedWidth={false}
          flip={null}
          icon="arrow-alt-circle-left"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="3x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <input
        className="AddFolder__input"
        id="post_input_field"
        onChange={[Function]}
        placeholder="folder title"
        type="text"
      />
      <button
        className="AddFolder__button"
        type="submit"
      >
        submit
      </button>
    </form>
  </div>
</Fragment>
`;

exports[`AddFolderForm component renders the AddFolderForm given props 1`] = `
<Fragment>
  <button
    className="go_back_button"
    onClick={[Function]}
  />
  <div
    className="AddFolder__form_container"
  >
    <form
      className="AddFolder__form"
      onSubmit={[Function]}
    >
      <button
        className="go_back_button"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon
          border={false}
          className="arrow-alt-circle-left"
          fixedWidth={false}
          flip={null}
          icon="arrow-alt-circle-left"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="3x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <input
        className="AddFolder__input"
        id="post_input_field"
        onChange={[Function]}
        placeholder="folder title"
        type="text"
      />
      <button
        className="AddFolder__button"
        type="submit"
      >
        submit
      </button>
    </form>
  </div>
</Fragment>
`;
